client/core/trade_simnet_test.go

```diff
@@ -238,6 +243,90 @@ func testNoMakerRedeem(t *testing.T) {
 	}
 }
 
+// testMakerGhostingAfterTakerRedeem places simple orders for clients 1 and 2,
+// neogiates the resulting trades smoothly till TakerSwapCast, then Maker goes
+// AWOL after redeeming taker's swap without notifying Taker. This test ensures
+// that Taker auto-finds Maker's redeem, extracts the secret key and redeems
+// Maker's swap to complete the trade.
+// A scenario where Maker actually did NOT redeem Taker's swap is handled in
+// testNoMakerRedeem which ensures that both parties are able to refund their
+// swaps.
+// TODO: What happens if FindRedemption encounters a refund instead of a redeem?
```

Comment:
We are the only ones who could have refunded it though, right?

Reply:
Right. If we have redeemed, FindRedemption would error when attempting to extract the contract secret from the spending input sig. Something I want to test out though.